html2ebooks/dn-html2ebooks.1
.\" Automatically generated by Pod::Man 4.09 (Pod::Simple 3.35)
.\"
.\" Standard preamble:
.\" ========================================================================
.de Sp \" Vertical space (when we can't use .PP)
.if t .sp .5v
.if n .sp
..
.de Vb \" Begin verbatim text
.ft CW
.nf
.ne \\$1
..
.de Ve \" End verbatim text
.ft R
.fi
..
.\" Set up some character translations and predefined strings.  \*(-- will
.\" give an unbreakable dash, \*(PI will give pi, \*(L" will give a left
.\" double quote, and \*(R" will give a right double quote.  \*(C+ will
.\" give a nicer C++.  Capital omega is used to do unbreakable dashes and
.\" therefore won't be available.  \*(C` and \*(C' expand to `' in nroff,
.\" nothing in troff, for use with C<>.
.tr \(*W-
.ds C+ C\v'-.1v'\h'-1p'\s-2+\h'-1p'+\s0\v'.1v'\h'-1p'
.ie n \{\
.    ds -- \(*W-
.    ds PI pi
.    if (\n(.H=4u)&(1m=24u) .ds -- \(*W\h'-12u'\(*W\h'-12u'-\" diablo 10 pitch
.    if (\n(.H=4u)&(1m=20u) .ds -- \(*W\h'-12u'\(*W\h'-8u'-\"  diablo 12 pitch
.    ds L" ""
.    ds R" ""
.    ds C` ""
.    ds C' ""
'br\}
.el\{\
.    ds -- \|\(em\|
.    ds PI \(*p
.    ds L" ``
.    ds R" ''
.    ds C`
.    ds C'
'br\}
.\"
.\" Escape single quotes in literal strings from groff's Unicode transform.
.ie \n(.g .ds Aq \(aq
.el       .ds Aq '
.\"
.\" If the F register is >0, we'll generate index entries on stderr for
.\" titles (.TH), headers (.SH), subsections (.SS), items (.Ip), and index
.\" entries marked with X<> in POD.  Of course, you'll have to process the
.\" output yourself in some meaningful fashion.
.\"
.\" Avoid warning from groff about undefined register 'F'.
.de IX
..
.if !\nF .nr F 0
.if \nF>0 \{\
.    de IX
.    tm Index:\\$1\t\\n%\t"\\$2"
..
.    if !\nF==2 \{\
.        nr % 0
.        nr F 2
.    \}
.\}
.\"
.\" Accent mark definitions (@(#)ms.acc 1.5 88/02/08 SMI; from UCB 4.2).
.\" Fear.  Run.  Save yourself.  No user-serviceable parts.
.    \" fudge factors for nroff and troff
.if n \{\
.    ds #H 0
.    ds #V .8m
.    ds #F .3m
.    ds #[ \f1
.    ds #] \fP
.\}
.if t \{\
.    ds #H ((1u-(\\\\n(.fu%2u))*.13m)
.    ds #V .6m
.    ds #F 0
.    ds #[ \&
.    ds #] \&
.\}
.    \" simple accents for nroff and troff
.if n \{\
.    ds ' \&
.    ds ` \&
.    ds ^ \&
.    ds , \&
.    ds ~ ~
.    ds /
.\}
.if t \{\
.    ds ' \\k:\h'-(\\n(.wu*8/10-\*(#H)'\'\h"|\\n:u"
.    ds ` \\k:\h'-(\\n(.wu*8/10-\*(#H)'\`\h'|\\n:u'
.    ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'^\h'|\\n:u'
.    ds , \\k:\h'-(\\n(.wu*8/10)',\h'|\\n:u'
.    ds ~ \\k:\h'-(\\n(.wu-\*(#H-.1m)'~\h'|\\n:u'
.    ds / \\k:\h'-(\\n(.wu*8/10-\*(#H)'\z\(sl\h'|\\n:u'
.\}
.    \" troff and (daisy-wheel) nroff accents
.ds : \\k:\h'-(\\n(.wu*8/10-\*(#H+.1m+\*(#F)'\v'-\*(#V'\z.\h'.2m+\*(#F'.\h'|\\n:u'\v'\*(#V'
.ds 8 \h'\*(#H'\(*b\h'-\*(#H'
.ds o \\k:\h'-(\\n(.wu+\w'\(de'u-\*(#H)/2u'\v'-.3n'\*(#[\z\(de\v'.3n'\h'|\\n:u'\*(#]
.ds d- \h'\*(#H'\(pd\h'-\w'~'u'\v'-.25m'\f2\(hy\fP\v'.25m'\h'-\*(#H'
.ds D- D\\k:\h'-\w'D'u'\v'-.11m'\z\(hy\v'.11m'\h'|\\n:u'
.ds th \*(#[\v'.3m'\s+1I\s-1\v'-.3m'\h'-(\w'I'u*2/3)'\s-1o\s+1\*(#]
.ds Th \*(#[\s+2I\s-2\h'-\w'I'u*3/5'\v'-.3m'o\v'.3m'\*(#]
.ds ae a\h'-(\w'a'u*4/10)'e
.ds Ae A\h'-(\w'A'u*4/10)'E
.    \" corrections for vroff
.if v .ds ~ \\k:\h'-(\\n(.wu*9/10-\*(#H)'\s-2\u~\d\s+2\h'|\\n:u'
.if v .ds ^ \\k:\h'-(\\n(.wu*10/11-\*(#H)'\v'-.4m'^\v'.4m'\h'|\\n:u'
.    \" for low resolution devices (crt and lpr)
.if \n(.H>23 .if \n(.V>19 \
\{\
.    ds : e
.    ds 8 ss
.    ds o a
.    ds d- d\h'-1'\(ga
.    ds D- D\h'-1'\(hy
.    ds th \o'bp'
.    ds Th \o'LP'
.    ds ae ae
.    ds Ae AE
.\}
.rm #[ #] #H #V #F C
.\" ========================================================================
.\"
.IX Title "DN-HTML2EBOOKS 1"
.TH DN-HTML2EBOOKS 1 "2017-10-03" "perl v5.26.0" "User Contributed Perl Documentation"
.\" For nroff, turn off justification.  Always turn off hyphenation; it makes
.\" way too many mistakes in technical documents.
.if n .ad l
.nh
.SH "NAME"
dn\-html2ebooks \- convert html file to ebook formats
.SH "USAGE"
.IX Header "USAGE"
\&\fBdn\-html2ebooks\fR \fB\-b\fR \fIbasename\fR \fB\-t\fR \fItitle\fR \fB\-a\fR \fIauthor\fR
.PP
\&\fBdn\-html2ebooks \-h\fR
.SH "REQUIRED OPTIONS"
.IX Header "REQUIRED OPTIONS"
.IP "\fB\-b\fR  \fB\-\-basename\fR \fIbasename\fR" 4
.IX Item "-b --basename basename"
Basename (file name without extension) of source html file.
.Sp
Scalar string. Required.
.IP "\fB\-t\fR  \fB\-\-title\fR \fItitle\fR" 4
.IX Item "-t --title title"
Title of book. Enclose in quotes if it contains spaces.
.Sp
Scalar string. Required.
.IP "\fB\-a\fR  \fB\-\-author\fR \fIauthor\fR" 4
.IX Item "-a --author author"
Author (or authors) of book. Enclose in quotes if it contains spaces.
.Sp
Scalar string. Required.
.SH "DESCRIPTION"
.IX Header "DESCRIPTION"
Converts an html file in the current directory named \fIbasename.html\fR or
\&\fIbasename.htm\fR where \*(L"basename\*(R" is the option provided to the \f(CW\*(C`\-b\*(C'\fR option.
This source file is converted to the following formats and output files:
.IP "Mobipocket e\-book (\fIbasename.mobi\fR)" 4
.IX Item "Mobipocket e-book (basename.mobi)"
.PD 0
.IP "Electronic publication (\fIbasename.epub\fR)" 4
.IX Item "Electronic publication (basename.epub)"
.IP "Kindle Format 8 (\fIbasename.epub\fR)" 4
.IX Item "Kindle Format 8 (basename.epub)"
.PD
.PP
Output files are written to the current directory and silently overwrite any
existing output files of the same name.
.PP
If there is a png image file in the current directory called \fIbasename.png\fR it
will be used as a cover image for the ebooks.
.PP
The conversions are performed by \fIebook-convert\fR, part of the Calibre suite on
debian systems.
.SH "DEPENDENCIES"
.IX Header "DEPENDENCIES"
.SS "Perl modules"
.IX Subsection "Perl modules"
Carp, Curses, Dn::Role::HasPath, Dn::Role::HasString, Function::Parameters,
IPC::Cmd, Moo, MooX::HandlesVia, MooX::Options, namespace::clean, Readonly,
strictures, String::ShellQuote, Types::Standard.
.SS "Executables"
.IX Subsection "Executables"
ebook-convert.
.SH "BUGS AND LIMITATIONS"
.IX Header "BUGS AND LIMITATIONS"
Please report any bugs to the author.
.SH "AUTHOR"
.IX Header "AUTHOR"
David Nebauer (david at nebauer dot org)
.SH "LICENSE AND COPYRIGHT"
.IX Header "LICENSE AND COPYRIGHT"
Copyright (c) 2017 David Nebauer (david at nebauer dot org)
.PP
This script is free software; you can redistribute it and/or modify it under
the same terms as Perl itself.


texdepend/Changes
Revision history for Perl script texdepend.

0.91  Fri Jun 25 14:29:00 1999
	- updated version; created by sdist 0.3
	- fixed stupid error in $graphics_pat

0.92  13 Jul 1999 14:50:42
   - Added .cls files to @packages; added -styles option to append
     extensions of @styles files parsed in the .log.  Fixed small
     bug re -ignore option. [Thanks to Bernd Schandl
     <bschand@math.clemson.edu> for suggestions.]





html2ebooks/dn-html2ebooks.sh
#!/bin/bash

# File: dn-html2ebooks
# Author: David Nebauer (david at nebauer dot org)
# Purpose: Convert an html file to mobi, epub and azw3 formats
# Created: 2017-07-02

# VARIABLES

self='dn-html2ebooks'
parameters='-b basename -t title -a author'
divider='-----------------------'
basename=''
title=''
author=''

# PROCEDURES

# Show usage
#   params: nil
#   prints: nil
#   return: nil
displayUsage () {
cat << _USAGE
${self} converts an html file to ebook formats

Converts html file named "\${basename}.html" in the current
directory to mobi ("\${basename}.mobi"), epub
(\${basename}.epub) and azw3 (\${basename}.azw3) formats.

Output files are written to the current directory and
silently overwrite any existing output files of the same
name.

Requires cover image named "\${basename}.png" in the
current directory.

Usage: ${self} ${parameters}
       ${self} -h

Options: -b basename = basename of input and output files
         -t title    = title of book
         -a author   = author(s) of book
_USAGE
}
# Process command line
#   params: all command line parameters
#   prints: feedback
#   return: nil
processCommandLine () {
	# Read the command line options
	#   - if optstring starts with ':' then error reporting is suppressed
	#     leave ':' at start as '\?' and '\:' error capturing require it
	#   - if option is followed by ':' then it is expected to have an argument
	while getopts ":hb:t:a:" opt ; do
		case ${opt} in
			'h' ) displayUsage && exit 0;;
			'b' ) basename="${OPTARG}";;
			't' ) title="${OPTARG}";;
			'a' ) author="${OPTARG}";;
			'?' ) echo -e "Error: Invalid flag '${OPTARG}' detected"
				  echo -e "Usage: ${self} ${parameters}"
				  echo -e "Try '${self} -h' for help"
				  echo -ne "\a"
				  exit 1;;
			':' ) echo -e "Error: No argument supplied for flag '${OPTARG}'"
				  echo -e "Usage: ${self} ${parameters}"
				  echo -e "Try '${self} -h' for help"
				  echo -ne "\a"
				  exit 1;;
		esac
	done
	shift $((OPTIND-1))
	unset parameters
}


# MAIN

# Process command line
processCommandLine "${@}"

# Need ebook-convert
if ! which ebook-convert &>/dev/null ; then
    echo "${self}: Cannot find 'ebook-convert' program" > /dev/stderr
    echo "${self}: Script aborted" > /dev/stderr
    exit
fi

# Check arguments
proceed=true
# - basename
if test -n "${basename}" ; then
    if ! test -r "${basename}.html" ; then
        msg="${self}: Cannot read file '${basename}.html' in current directory"
        echo "${msg}" > /dev/stderr
        proceed=false
    fi
    if ! test -r "${basename}.png" ; then
        msg="${self}: Cannot read file '${basename}.png' in current directory"
        echo "${msg}" > /dev/stderr
        proceed=false
    fi
else
    echo "${self}: No basename provided" > /dev/stderr
    proceed=false
fi
# - title
if test -z "${title}" ; then
    echo "${self}: No book title provided" > /dev/stderr
    proceed=false
fi
# - author
if test -z "${author}" ; then
    echo "${self}: No book author(s) provided" > /dev/stderr
    proceed=false
fi
if test "${proceed}" = false ; then
    echo "${self}: Script aborted" > /dev/stderr
    exit
fi

# generate output files
# - mobi
echo "${divider}"
echo 'Converting to MOBI'
echo "${divider}"
ebook-convert \
    "${basename}.html" \
    "${basename}.mobi" \
    --no-inline-toc \
    --pretty-print \
    --smarten-punctuation \
    --insert-blank-line \
    --keep-ligatures \
    --title="${title}" \
    --authors="${author}" \
    --language='en_AU' \
    --pubdate="$(date -I)" \
    --cover="${basename}.png"
if test "${?}" = false ; then
    echo "Conversion to mobi format exited with error status" > dev/stderr
    echo "Script aborted" > /std/stderr
    exit
fi

# - epub
echo "${divider}"
echo 'Converting to EPUB'
echo "${divider}"
ebook-convert \
    "${basename}.html" \
    "${basename}.epub" \
    --pretty-print \
    --smarten-punctuation \
    --insert-blank-line \
    --keep-ligatures \
    --title="${title}" \
    --authors="${author}" \
    --language='en_AU' \
    --pubdate="$(date -I)" \
    --cover="${basename}.png"
if test "${?}" = false ; then
    echo "Conversion to epub format exited with error status" > dev/stderr
    echo "Script aborted" > /std/stderr
    exit
fi

# - azw3
echo "${divider}"
echo 'Converting to AZW3'
echo "${divider}"
ebook-convert \
    "${basename}.html" \
    "${basename}.azw3" \
    --no-inline-toc \
    --pretty-print \
    --smarten-punctuation \
    --insert-blank-line \
    --keep-ligatures \
    --title="${title}" \
    --authors="${author}" \
    --language='en_AU' \
    --pubdate="$(date -I)" \
    --cover="${basename}.png"
if test "${?}" = false ; then
    echo "Conversion to azw3 format exited with error status" > dev/stderr
    echo "Script aborted" > /std/stderr
    exit
fi

echo "${divider}"
echo 'Conversions complete'
